File: app/src/main/java/net/squanchy/venue/VenueInfoModule.java
package net.squanchy.venue;

import net.squanchy.service.firebase.FirebaseAuthService;
import net.squanchy.service.repository.VenueRepository;

import dagger.Module;
import dagger.Provides;

@Module
class VenueInfoModule {

    @Provides
    VenueInfoService favoritesService(VenueRepository venueRepository, FirebaseAuthService authService) {
        return new VenueInfoService(venueRepository, authService);
    }
}

File: app/src/main/java/net/squanchy/venue/VenueInfoModule.kt
package net.squanchy.venue

import dagger.Module
import dagger.Provides
import net.squanchy.service.firebase.FirestoreDbService
import net.squanchy.service.repository.AuthService

@Module
internal class VenueInfoModule {

    @Provides
    fun favoritesService(dbService: FirestoreDbService, authService: AuthService) = VenueInfoService(dbService, authService)
}
